web/hooks/use-contracts.ts

```diff
@@ -168,3 +170,28 @@ export const useContracts = (
 ) => {
   return useStoreItems(filterDefined(contractIds), listenForContract, options)
 }
+
+export function usePrivateContract(contractSlug: string, delay: number) {
+  const [privateContract, setPrivateContract] = useState<
+    Contract<AnyContractType> | undefined | null
+  >(undefined)
+  const user = useUser()
+  useEffect(() => {
+    // if there is no user
+    if (user === null) {
+      setPrivateContract(null)
+    } else if (user) {
+      // need this timeout (1 sec works) or else get "must be signed in to make API calls" error
```

Comment:
Hmm, would be worth investigating at some point why this happens.

Oh, I think it could be because we load the user from localstorage optimistically. The actual auth state is set as a callback to `onIdTokenChanged` in `auth-context.ts`.

One solution would be to add another field to the AuthUser state like `authLoaded: boolean`. Then, only set it once the callback to onIdTokenChanged is called. And maybe have a hook `useAuthUser` that gives the whole object.

If you don't want to do this, I can, after you merge.

Reply:
Please do this!